src/gt_pro.cpp

```diff
@@ -366,38 +388,27 @@ int main(int argc, char** argv) {
 	// char seq_buf[k+1];
 
 	uint64_t start = -1;
-	uint64_t end = 0;
-
-	unordered_map<uint32_t, tuple<uint64_t, uint64_t>> lmer_indx;
+	uint64_t end = 0;  // FIXME: this init breaks the loop invariant;  should be -1
 
 	vector<uint32_t> mmers;
 	vector<uint64_t> snps;
-	uint32_t cur_lmer = 2147483648;
-
-	/* endianness test
-	   bool is_lmer = false;
-
-	   char t_lmer_buf[l+1];
-	   char t_mmer_buf[m+1];
-	   for (int i = 0; i < filesize/4; i++) {
-	// little endian
-	if (!is_lmer) {
-	seq_decode<uint32_t>(t_mmer_buf, m+1, mmappedData[i], b_mask);
-	is_lmer = true;
-	} else {
-	seq_decode<uint32_t>(t_lmer_buf, l+1, mmappedData[i], b_mask);
-	is_lmer = false;
-	cerr << chrono_time() << ":  " << t_lmer_buf << t_mmer_buf << '\n';
-	}
-	}
-
-	return 0;
-	*/
+	uint32_t cur_lmer = 2147483648;  // 1u << 31u
 
 	auto l_start = chrono_time();
 
+	cerr << chrono_time() << ":  " << "[OK] Reserving memory." << endl;
+	mmers.reserve(filesize / 8);
+	snps.reserve(filesize / 8);
+
+	cerr << chrono_time() << ":  " << "[OK] Zeroing index." << endl;
+
+	// unordered_map<uint32_t, tuple<uint64_t, uint64_t>> lmer_indx;
+	constexpr uint64_t k_4_BILLION = ((uint64_t) 1) << (uint64_t) 32;
```

Comment:
this can probably be 1 billion, considering L=15

just need to fix the mast below from 0xFFFFFFFF to 0x3FFFFFFF